### PR TITLE
Wp 535/tron consolidation UI

### PR DESCRIPTION
**Pull request by mohammadalfaiyazbitgo**
Screenshot 2023-09-01 at 2 56 28 PM
Screenshot 2023-09-01 at 2 55 54 PM

https://github.com/BitGo/wallet-recovery-wizard/assets/101133667/99e49596-aa72-4a1c-a4c7-025f9ddf68ea

### CODE REVIEW

**Comment by alia-bitgo** [comment]
Could you add in maybe a screenshot of what the form looks like?

File: electron/main/index.ts [hunk 1]
```diff
@@ -235,6 +235,20 @@ async function createWindow() {
...
+      case 'trx':
+      case 'ttrx': {
+        // Batch Consolidations only implemented for Tron for now so have
+        // to reference Trx Coin directly instead of Basecoin;
+        const trxCoin = sdk.coin(coin) as Trx | Ttrx;
+        return await trxCoin.recoverConsolidations(params);
+      }
+      default:
+        return new Error(`Coin: ${coin} does not support consolidation recovery`);
+    }
```
Comment (zahin-mohammad) [inline electron/main/index.ts:249]: can you wrap this in a try/catch and return the error to the UI thread?

Comment (zahin-mohammad) [reply electron/main/index.ts:249]: Pretty sure errors in the icp handlers are swallowed by the nodejs process and doesnt surface to the UI

**Review by zahin-mohammad — DISMISSED** [review]
non-blocking

**Comment by zahin-mohammad** [comment]
> Could you add in maybe a screenshot of what the form looks like?

+1, pleas add screenshots for new UI changes, or a small video if thats easier.

File: electron/preload/index.ts [hunk 2]
```diff
@@ -10,13 +10,19 @@ import type {
...
 import { contextBridge, ipcRenderer } from 'electron';
 import type { ObjectEncodingOptions } from 'node:fs';
 
 type User = { username: string };
 
 type Commands = {
+  recoverConsolidations(
+    coin: string,
+    params: ConsolidationRecoveryOptions
+  ): Promise<ConsolidationRecoveryBatch>;
```
Comment (zahin-mohammad) [inline electron/preload/index.ts:25]: This return type needs to be updated to have the error as well

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]

File: src/containers/BuildUnsignedConsolidation/TronForm.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            HelperText="The starting index of addresses to consolidate"
+            Label="Starting Scan Index"
+            name="startingScanIndex"
+            Width="fill"
+          />
+        </div>
+        <div className="tw-mb-4">
+          <FormikTextfield
+            HelperText="The ending index of addresses to consolidate"
+            Label="Ending Scan Index"
```
Comment (evanzbitgo) [inline src/containers/BuildUnsignedConsolidation/TronForm.tsx:100]: Could you add `(exclusive)`?

File: src/containers/BuildUnsignedConsolidation/TronForm.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            HelperText="The BitGo public key for the wallet, as found on your recovery KeyCard."
+            Label="BitGo Public Key"
+            name="bitgoKey"
+            Width="fill"
+          />
+        </div>
+        <div className="tw-mb-4">
+          <FormikTextfield
+            HelperText="The starting index of addresses to consolidate"
+            Label="Starting Scan Index"
```
Comment (evanzbitgo) [inline src/containers/BuildUnsignedConsolidation/TronForm.tsx:92]: `(inclusive)`

File: electron/preload/index.ts [hunk 5]
```diff
@@ -103,6 +109,9 @@ const queries: Queries = {
 };
 
 const commands: Commands = {
+  recoverConsolidations(coin: string, params: ConsolidationRecoveryOptions): Promise<ConsolidationRecoveryBatch> {
```
Comment (zahin-mohammad) [inline electron/preload/index.ts:112]: This type also needs to be updated.

Comment (zahin-mohammad) [inline electron/preload/index.ts:112]:
```suggestion
  recoverConsolidations(coin: string, params: ConsolidationRecoveryOptions): Promise<Error | ConsolidationRecoveryBatch> {
```

Comment (mohammadalfaiyazbitgo) [reply electron/preload/index.ts:112]: it's there?

**Review by zahin-mohammad — APPROVED** [review]

**Review by evanzbitgo — APPROVED** [review]

**Review by alia-bitgo — APPROVED** [review]

**Review by alia-bitgo — APPROVED** [review]

**Review by evanzbitgo — APPROVED** [review]

**Pull request merged**